compiler/passes/insertLineNumbers.cpp

```diff
@@ -314,8 +344,21 @@ static void moveLinenoInsideArgBundle()
         Expr* fileActual = call->argList.tail->remove();
         Expr* lineActual = call->argList.tail->remove();
         Expr* bundleActual = call->argList.tail;
-        call->insertBefore(new CallExpr(PRIM_SET_MEMBER, bundleActual->copy(), lineField, lineActual));
-        call->insertBefore(new CallExpr(PRIM_SET_MEMBER, bundleActual->copy(), fileField, fileActual));
+        call->insertBefore(new CallExpr(PRIM_SET_MEMBER,
+              bundleActual->copy(), lineField, lineActual));
+
+        // As above, we cant use the file directly with the bundle
+        // Instead we will perform a lookup to find it's index in the table and
+        // pass that along instead. If we are using C line numbers we dont need
+        // to pass anything.
+        if (!fCLineNumbers) {
+          VarSymbol* fileIdxTemp = newTemp("fnIdx_tmp", dtUInt[INT_SIZE_64]);
+          call->insertBefore(new DefExpr(fileIdxTemp));
+          call->insertBefore(new CallExpr(PRIM_MOVE, fileIdxTemp,
+                new CallExpr(PRIM_FIND_FILENAME_IDX, fileActual)));
```

Comment:
We do not know what the file name is at compile time, they are passed through to functions so we can try and give a better error location for the user.